### PR TITLE
VReplication: Test Migrations From MariaDB to MySQL

### PR DESCRIPTION
**Pull request by mattlord**
## Description

We [deprecated formal support for MariaDB in Vitess v14+](https://github.com/vitessio/vitess/issues/9518) but we always want to have a migration path into Vitess for current MariaDB users so they should be able to use VReplication's [`MoveTables`](https://vitess.io/docs/reference/vreplication/movetables/)/[`Migrate`](https://vitess.io/docs/16.0/reference/vreplication/migrate/) workflows for this purpose. The new `TestMoveTablesMariaDBToMySQL` end-to-end test ensures that we continue to support that use case.

In this PR we consolidate all of the MariaDB 10.x config files — there were no substantial config differences between them — into a single MariaDB 10 config file and we modify [`mysqlctl`](https://vitess.io/docs/16.0/reference/programs/mysqlctl/) to use this single config file anytime MariaDB 10 is detected ([MariaDB 11.0 was just announced](https://mariadb.org/mariadb-11-0-new-optimizer/)). This was needed as the test uses the [latest GA release of MariaDB available today](https://mariadb.org/download/): 10.10.2.

We also remove the MariaDB 10.3 unit test here as [10.3 will shortly be EOL](https://mariadb.org/about/#maintenance-policy) and this new end-to-end test confirms the key bit of support that we must ensure that we maintain in v16+.

## Related Issue(s)

  - Fixes: https://github.com/vitessio/vitess/issues/10732
  - Related to: https://github.com/vitessio/vitess/pull/11985

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added 
-   [x] Documentation is not required

### CODE REVIEW

**Review by dbussink — APPROVED** [review]
I really like the cleanup here and also makes dealing with #11985 much easier, so I'm all for it :smile:. 

> to use this single config file anytime MariaDB 10 is detected ([MariaDB 11.0 was just announced](https://mariadb.org/mariadb-11-0-new-optimizer/))

@mattlord Do we have any idea if the MariaDB 11.0 release would significantly impact our migration path here? Or is the underlying stuff still similar enough that it will still work? 

I guess that once the official MariaDB 11.0 comes out, we'll want to add that to these tests as well?

**Comment by dbussink** [comment]
> #### If a workflow is added or modified:

Do we need to update any settings for the removed MariaDB workflow for the unit tests? I think we also can / should cleanup `test/templates/unit_test.tpl` and `test/ci_workflow_gen.go` for MariaDB references?

**Comment by dbussink** [comment]
> Do we need to update any settings for the removed MariaDB workflow for the unit tests? I think we also can / should cleanup `test/templates/unit_test.tpl` and `test/ci_workflow_gen.go` for MariaDB references?

@mattlord Took the liberty to push up changes to clean this up. If we don't want that, feel free to force push it away :smile:.

File: test.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ const (
 	configFileName = "test/config.json"
 
 	// List of flavors for which a bootstrap Docker image is available.
-	flavors = "mysql56,mysql57,mysql80,mariadb,mariadb103,percona,percona57,percona80"
+	flavors = "mysql56,mysql57,mysql80,percona,percona57,percona80"
```
[the remark this answers is not in the document]
Comment (frouioui) [reply test.go:113]: We should since we no longer use `mysql56` in our GitHub Actions workflows

[the remark this answers is not in the document]
Comment (dbussink) [reply test.go]: Cleaned up that one as well!

**Review by frouioui — APPROVED** [review]
Nothing special to say, everything looks good to me!

File: go/vt/mysqlctl/mysqld.go [hunk 2]
```diff
@@ -217,9 +217,7 @@ func GetVersionFromEnv() (flavor MySQLFlavor, ver ServerVersion, err error) {
 	env := os.Getenv("MYSQL_FLAVOR")
 	switch env {
 	case "MariaDB":
-		return FlavorMariaDB, ServerVersion{10, 0, 10}, nil
-	case "MariaDB103":
```
Comment (deepthi) [inline go/vt/mysqlctl/mysqld.go:221]: This is fine, given #9518.

**Review by deepthi — APPROVED** [review]
Nice work. LGTM

**Comment by dbussink** [comment]
Found some more things we need to clean up, see https://github.com/vitessio/vitess/pull/12040

**Pull request merged**